pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go

```diff
@@ -95,6 +95,181 @@ func getApiURL(resourceOrWorkspace string, isAppInsightsQuery bool) string {
 	}
 }
 
+func buildLogAnalyticsQuery(query backend.DataQuery, dsInfo types.DatasourceInfo, appInsightsRegExp *regexp.Regexp) (*AzureLogAnalyticsQuery, error) {
+	queryJSONModel := types.LogJSONQuery{}
+	err := json.Unmarshal(query.JSON, &queryJSONModel)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode the Azure Log Analytics query object from JSON: %w", err)
+	}
+	resources := []string{}
+	var resourceOrWorkspace string
+	var queryString string
+	resultFormat := dataquery.ResultFormat(types.TimeSeries)
+	appInsightsQuery := false
+	dashboardTime := false
+	timeColumn := ""
+	azureLogAnalyticsTarget := queryJSONModel.AzureLogAnalytics
+
+	if azureLogAnalyticsTarget.ResultFormat != nil {
+		resultFormat = *azureLogAnalyticsTarget.ResultFormat
+	}
+
+	// Legacy queries only specify a Workspace GUID, which we need to use the old workspace-centric
+	// API URL for, and newer queries specifying a resource URI should use resource-centric API.
+	// However, legacy workspace queries using a `workspaces()` template variable will be resolved
+	// to a resource URI, so they should use the new resource-centric.
+	if len(azureLogAnalyticsTarget.Resources) > 0 {
```

Comment:
This block could be broken out into 2-3 separate functions, for handling `resources`, `resourceOrWorkspace` and `appInsightsQuery` separately.

Reply:
Have broken this out into a helper that'll return `resources` and `resourceOrWorkspace`. Given that `appInsightsQuery` is a single call I don't think it warranted its own function but let me know what you think 😊